sonic-chassisd/scripts/chassisd

```diff
@@ -603,6 +808,52 @@ class ConfigManagerTask(ProcessTaskBase):
 
             self.config_updater.module_config_update(key, admin_state)
 
+
+#
+# SmartSwitch Config Manager task ========================================================
+#
+
+
+class SmartSwitchConfigManagerTask(ProcessTaskBase):
+    def __init__(self):
+        ProcessTaskBase.__init__(self)
+
+        # TODO: Refactor to eliminate the need for this Logger instance
+        self.logger = logger.Logger(SYSLOG_IDENTIFIER)
+
+    def task_worker(self):
+        self.config_updater = SmartSwitchModuleConfigUpdater(SYSLOG_IDENTIFIER, platform_chassis)
+        config_db = daemon_base.db_connect("CONFIG_DB")
+
+        # Subscribe to CHASSIS_MODULE table notifications in the Config DB
+        sel = swsscommon.Select()
+        sst = swsscommon.SubscriberStateTable(config_db, CHASSIS_CFG_TABLE)
+        sel.addSelectable(sst)
+
+        # Listen indefinitely for changes to the CFG_CHASSIS_MODULE_TABLE table in the Config DB
+        while True:
+            # Use timeout to prevent ignoring the signals we want to handle
+            # in signal_handler() (e.g. SIGTERM for graceful shutdown)
+            (state, c) = sel.select(SELECT_TIMEOUT)
+
+            if state == swsscommon.Select.TIMEOUT:
+                # Do not flood log when select times out
+                continue
+            if state != swsscommon.Select.OBJECT:
+                self.logger.log_warning("sel.select() did not return swsscommon.Select.OBJECT")
+                continue
+
+            (key, op, fvp) = sst.pop()
+
+            if op == 'SET':
```

Comment:
@rameshraghupathy SET operation can set ADMIN down or UP. please parse the key to know if the user is setting admin UP/DOWN

Reply:
> @rameshraghupathy SET operation can set ADMIN down or UP. please parse the key to know if the user is setting admin UP/DOWN

@prgeor Good question.  The original behavior is: The default is no config - meaning admin up   when the user wants to shutdown a module adds a config "admins_status: down".  So, the SET operation has been used for "admins_status: down" and DEL has been used for setting the module up.   If you want we can change this for Smartswitch.  Please let me know.

Reply:
@rameshraghupathy we need to handle following scenarios:-

1. At run time if user configures module up/down then process the event
2. When chassisd is booting up and if the admin status is missing in config DB then process to keep the DPU in dark mode. In this case there will not be any event triggered. Its a DB read. 
3. While chassisd bootup if the config DB already has admin UP/DOWN then process accordingly

@rameshraghupathy please make sure if chasssid crashes at run time it does not bring down the DPU if DPU was already admin up.

Reply:
> @rameshraghupathy we need to handle following scenarios:-
> 
> 1. At run time if user configures module up/down then process the event
> 2. When chassisd is booting up and if the admin status is missing in config DB then process to keep the DPU in dark mode. In this case there will not be any event triggered. Its a DB read.
> 3. While chassisd bootup if the config DB already has admin UP/DOWN then process accordingly
> 
> @rameshraghupathy please make sure if chasssid crashes at run time it does not bring down the DPU if DPU was already admin up.

@prgeor Support for (1) exists in the original implementation. (2) and (3) are handled by  "def set_initial_dpu_admin_state(self):"

Reply:
> @rameshraghupathy please make sure if chasssid crashes at run time it does not bring down the DPU if DPU was already admin up.

@prgeor Already verified this case

Reply:
This code needs to be updated as per discussion, We are writing the admin state data (irrespective of up/down) to the CONFIG_DB from sonic-utilities, but here we are checking to see if data is set or removed (instead of checking the admin state value (as per previous discussion)

Reply:
@gpunathilell  Fixed